src/BackgroundTask/CronBasedBackgroundTask.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * Abstract class CronBasedBackgroundTask.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\BackgroundTask;
+
+use AmpProject\AmpWP\HasDeactivation;
+use AmpProject\AmpWP\Service;
+use AmpProject\Fonts;
+
+/**
+ * Abstract base class for using cron to execute a background task.
+ *
+ * @package AmpProject\AmpWP
+ */
+abstract class CronBasedBackgroundTask implements Service, HasDeactivation {
+
+	const DEFAULT_INTERVAL_HOURLY      = 'hourly';
+	const DEFAULT_INTERVAL_TWICE_DAILY = 'twicedaily';
+	const DEFAULT_INTERVAL_DAILY       = 'daily';
+
+	/**
+	 * List of default interval names that are shipped with WordPress.
+	 *
+	 * @var string[]
+	 */
+	const DEFAULT_INTERVAL_NAMES = [
```

Comment:
This is currently unused.

Reply:
Ah, good catch. When the cron-handling was still a bit more complex, this was used for validation. However, I ripped all of that out, and just assume someone using the class is using it correctly for now.

So this one can go.